x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/api/ml_models/create_ml_inference_pipeline.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { createApiLogic } from '../../../shared/api_logic/create_api_logic';
+import { HttpLogic } from '../../../shared/http';
+
+export interface CreateMlInferencePipelineApiLogicArgs {
+  indexName: string;
+  pipelineName: string;
+  modelId: string;
+  sourceField: string;
+  destinationField?: string;
+}
+
+export const createMlInferencePipeline = async (
```

Comment:
@demjened I went ahead and optimistically implemented this, and we can update as needed.